File: src/test/java/im/tox/tox4j/core/exceptions/ToxGetPortExceptionTest.java
package im.tox.tox4j.core.exceptions;

import im.tox.tox4j.ToxCoreTestBase;
import im.tox.tox4j.core.ToxCore;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class ToxGetPortExceptionTest extends ToxCoreTestBase {

  @Test
  public void testGetTcpPort_NotBound() throws Exception {
    try (ToxCore tox = newTox()) {
      tox.getTcpPort();
      fail();
    } catch (ToxGetPortException e) {
      assertEquals(ToxGetPortException.Code.NOT_BOUND, e.code());
    }
  }

}

File: projects/tox4j/src/test/java/im/tox/tox4j/core/exceptions/ToxGetPortExceptionTest.scala
package im.tox.tox4j.core.exceptions

import im.tox.tox4j.testing.ToxTestMixin
import org.scalatest.FunSuite

final class ToxGetPortExceptionTest extends FunSuite with ToxTestMixin {

  test("GetTcpPort_NotBound") {
    interceptWithTox(ToxGetPortException.Code.NOT_BOUND)(
      _.getTcpPort
    )
  }

}
